File: delivery-system-server/src/main/java/ru/delivery/system/model/json/product/ProductListResponse.java
package ru.delivery.system.model.json.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

public class ProductListResponse {
    @JsonProperty(value = "productList")
    @Getter
    @Setter
    private List<Product> productList;

    public static class Product {
        @JsonProperty(value = "productId")
        @Getter @Setter
        private Integer productId;

        @JsonProperty(value = "count")
        @Getter @Setter
        private Integer count;

        @JsonProperty(value = "cost")
        @Getter @Setter
        private Double cost;

        @JsonProperty("name")
        @Getter @Setter
        private String name;

        @JsonProperty("category")
        @Getter @Setter
        private String category;

        @JsonProperty("weight")
        @Getter @Setter
        private Double weight;

        @JsonProperty("height")
        @Getter @Setter
        private Double height;

        @JsonProperty("width")
        @Getter @Setter
        private Double width;

        @JsonProperty("length")
        @Getter @Setter
        private Double length;
    }
}

File: delivery-system-desktop/src/main/kotlin/ru/delivery/system/models/json/ProductListResponse.kt
package ru.delivery.system.models.json

import com.fasterxml.jackson.annotation.JsonProperty

class ProductListEntity {
    @JsonProperty("productList")
    var productList: List<Product>? = null

    class Product {
        @JsonProperty("productId")
        var productId: Int? = null

        @JsonProperty("count")
        var count: Int? = null

        @JsonProperty("cost")
        var cost: Double? = null

        @JsonProperty("name")
        var name: String? = null

        @JsonProperty("category")
        var category: String? = null

        @JsonProperty("weight")
        var weight: Double? = null

        @JsonProperty("height")
        var height: Double? = null

        @JsonProperty("width")
        var width: Double? = null

        @JsonProperty("length")
        var length: Double? = null
    }
}
